exercicios/para-casa/Sayonara/SAYONARA.sistema_de_alunas.py

```diff
@@ -0,0 +1,152 @@
+from dataset_alunas import dataset
+
+def main():
+    print("\n---  Seja bem vinda a Escola do Reprograma!  ---")
+    print("Sistema de informações de alunas")
+    
+    while True:
+        cod_opcao = obter_opcao()
+        
+        if cod_opcao == 1: incluir_nova_aluna()
+        elif cod_opcao == 2: consultar_lista_alunas(dataset)
+        elif cod_opcao == 3: consultar_faltas_aluna()
+        elif cod_opcao == 4: consultar_notas_aluna()
+        elif cod_opcao == 5: consultar_status_aprovacao()
+        elif cod_opcao == 6: print("Encerrando o programa..."); break
+
+def obter_opcao():
+    codigo_opcao = 0
+
+    while codigo_opcao not in [1, 2, 3, 4, 5, 6]:
+        try:
+            codigo_opcao = int(input("\nEscolha uma opção:\n"
+                                    "1 - Incluir uma nova aluna\n"
+                                    "2 - Consultar lista de alunas\n"
+                                    "3 - Consultar faltas da aluna\n"
+                                    "4 - Consultar notas da aluna\n"
+                                    "5 - Consultar status de aprovação\n"
+                                    "6 - Sair do sistema\n"
+                                    "Opção: "))
+                
+            if codigo_opcao not in [1, 2, 3, 4, 5, 6]:
+                print("Opção inválida. Por favor, escolha uma opção válida (1 a 5).\n")
+        except ValueError:
+            print("Entrada inválida. Por favor, digite um número inteiro.\n")
+            
+        return codigo_opcao
+    
+def incluir_nova_aluna():
+    print("Informe os dados ")
+    nome= input("Informe o nome e o sobrenome :")
+    turma = input("informe a turma :" )
+    notas = pedir_notas()
+    nota_participaçao=float(input("participação da aluna: "))
+    presença= chamada() 
+    dataset[(nome.split()[0] , nome.split()[1])] = {
+        "Turma": turma,
+         "Notas": notas,
+         "Presença": presença,
+         "Paticipaçao": nota_participaçao
```

Comment:
Repare que estamos usando uma chave aninhada chamada "Participaçao", e na função de consulta de aprovação, a chave que buscamos é "Participação" com o til. O que causa erro ao buscar dados de alunas que cadastramos posteriormente. É importante manter consistência com os dados que já temos cadastrados no dataset. 

**Sugestão:**
```suggestion
         "Paticipação": nota_participaçao
```